source/MaterialXCore/Interface.cpp

```diff
@@ -73,30 +74,63 @@ bool PortElement::validate(string* message) const
     NodePtr connectedNode = getConnectedNode();
     if (hasNodeName() || hasOutputString())
     {
-        validateRequire(connectedNode != nullptr, res, message, "Invalid port connection");
+        if (!connectedNode)
+        {
+            NodeGraphPtr nodeGraph = resolveRootNameReference<NodeGraph>(getNodeName());
+            if (!nodeGraph)
+            {
+                validateRequire(connectedNode != nullptr, res, message, "Invalid port connection");
+            }
+        }
```

Comment:
Could this proposed change perhaps be an error?  It looks like it's requiring that connectedNode be non-NULL after confirming that it's NULL.

Reply:
Yes, This looks like a debugging change. The if (!connectedNode) should be removed.